web/packages/teleport/src/Notifications/Notification.tsx

```diff
@@ -71,29 +70,47 @@ export function Notification({
         notificationState: NotificationState.CLICKED,
       })
       .then(res => {
-        setClicked(true);
+        markNotificationAsClicked(notification.id);
         return res;
       })
   );
 
   const [hideNotificationAttempt, hideNotification] = useAsync(() => {
-    return ctx.notificationService.upsertNotificationState(clusterId, {
-      notificationId: notification.id,
-      notificationState: NotificationState.DISMISSED,
-    });
+    return ctx.notificationService
+      .upsertNotificationState(clusterId, {
+        notificationId: notification.id,
+        notificationState: NotificationState.DISMISSED,
+      })
+      .then(() => {
+        removeNotification(notification.id);
+      });
   });
+
+  function onMarkAsClicked() {
+    if (notification.localNotification) {
+      ctx.storeNotifications.markNotificationAsClicked(notification.id);
+      markNotificationAsClicked(notification.id);
+      return;
```

Comment:
this at some point makes a network request to mark as clicked right? And the point of doing it here is to have the 'optimistic' UI instantly update while the network request is made in the background? If that is correct, where do we handle the "failed" update? Does this state get reset if the update fails?

Reply:
> this at some point makes a network request to mark as clicked right?

No, since this is for a `localNotification` (ie. just one from the frontend store, which isn't stored in the backend), there's no network call made and the notification state keeping track of it being clicked is solely in `localStorage`.